File: transaction-bff/src/main/java/br/com/felipe/gadelha/coffeandit/transactionbff/domain/repository/TransactionRepository.java
package br.com.felipe.gadelha.coffeandit.transactionbff.domain.repository;

import br.com.felipe.gadelha.coffeandit.transactionbff.domain.entity.Transaction;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface TransactionRepository {

    Flux<Transaction> findAll();

    Mono<Transaction> findById(String uuid);

    Mono<Long> save(Transaction transaction);

    Mono<Boolean> deleteAll();
}

File: transaction-ms/src/main/kotlin/br/com/felipe/gadelha/coffeandit/transactionms/domain/repository/TransactionRepository.kt
package br.com.felipe.gadelha.coffeandit.transactionms.domain.repository

import br.com.felipe.gadelha.coffeandit.transactionms.domain.entity.Transaction
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.stereotype.Repository

@Repository
interface TransactionRepository: ReactiveMongoRepository<Transaction, String>